tests/late_joiners.py

```diff
@@ -53,46 +53,48 @@ def find_primary(network):
     return term_info
 
 
-def timeout(node, suspend, election_timeout):
-    if suspend:
-        # We want to suspend the nodes' process so we need to initiate a new timer to wake it up eventually
-        node.suspend()
-        next_timeout = random.uniform(2 * election_timeout, 3 * election_timeout)
-        LOG.info(f"New timer set for node {node.node_id} is {next_timeout} seconds")
-        t = Timer(next_timeout, timeout, args=[node, False, 0])
-        t.start()
-    else:
-        node.resume()
-
-
-def assert_node_up_to_date(check, node, final_msg, final_msg_id):
+def assert_node_up_to_date(check, node, final_msg, final_msg_id, timeout=30):
     with node.user_client() as c:
-        for x in range(0, 5):
+        end_time = time.time() + timeout
+        while time.time() < end_time:
             try:
                 check(
                     c.get("LOG_get", {"id": final_msg_id}), result={"msg": final_msg},
                 )
                 return
-            except TimeoutError:
+            except (TimeoutError, requests.exceptions.ReadTimeout,) as e:
```

Comment:
Why do we need to specifically catch read timeout here? Is it because the enclave blocks completely during catch up?

Reply:
no, but there might be an early view change triggered and in progress, in which case we will not get responses to `LOG_get` requests right away

Reply:
It seems like we would want to increase the timeout then just swallowing the error

Reply:
Added comment here, pasting here for clarity:

        # Wait until final_msg_id is available in the node.
        # We need to catch timeout and assertion errors as
        # the node (e.g. late joiner) might not be up to date yet
        # or we might be in the middle of a view change so the node
        # will not be responsive. This timeout should be reduced when
        # checkpoints are implemented making catchup take less time